calcrule_validations/calculation_rule.py

```diff
@@ -49,40 +49,48 @@ def ready(cls):
 
     @classmethod
     def run_calculation_rules(
-        cls, sender, validation_class,
+        cls, sender, instance,
         user, context, **kwargs
     ):
         field_name = kwargs.pop('field_name')
         field_value = kwargs.pop('field_value')
         return cls.calculate_if_active_for_object(
-            validation_class,
+            instance,
             field_name=field_name,
             field_value=field_value,
             **kwargs
         )
 
     @classmethod
     def calculate_if_active_for_object(
-        cls, validation_class, calculation_uuid, **kwargs
+        cls, instance, calculation_uuid=None, **kwargs
     ):
+        if not calculation_uuid:
+            return False
         field_name = kwargs.pop('field_name')
         field_value = kwargs.pop('field_value')
-        if cls.active_for_object(validation_class, calculation_uuid):
-            return cls.calculate(validation_class, field_name, field_value, **kwargs)
+        if cls.active_for_object(instance, calculation_uuid):
+            return cls.calculate(instance, field_name, field_value, **kwargs)
 
     @classmethod
-    def active_for_object(cls, validation_class, calculation_uuid):
-        return cls.check_calculation(validation_class, calculation_uuid)
+    def active_for_object(cls, instance, calculation_uuid=None):
+        if not calculation_uuid:
+            return False
+        return cls.check_calculation(instance, calculation_uuid)
 
     @classmethod
     def get_linked_class(cls, sender, class_name, **kwargs):
         return ["Calculation"]
 
     @classmethod
-    def check_calculation(cls, validation_class, calculation_uuid, **kwargs):
-        return ValidationStrategyStorage.choose_strategy(validation_class).check_calculation(cls, calculation_uuid)
+    def check_calculation(cls, instance, calculation_uuid=None, **kwargs):
```

Comment:
Still not meeting what AbsStrategy requires: https://github.com/openimis/openimis-be-core_py/blob/develop/core/abs_calculation_rule.py#L97

Reply:
indeed we should have def check_calculation(cls, instance, *kwars) but if this calc is call with the abs def it will work 
let's say that I focused to make it work because of the urgency